src/Core/src/Extensions/ILayoutExtensions.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Microsoft.Maui.Extensions
+{
+	// TODO: Make this public on .NET 8
+	internal static class ILayoutExtensions
+	{
+		internal static void InvalidateChildrenIsEnabled(this ILayout layout)
+		{
+			foreach (var children in layout.GetChildren())
```

Comment:
`layout` is already an `IList<IView>`, so we don't need the GetChildren() method here. And this should be a `for` loop rather than a `foreach`, for performance reasons.